tests/lib/ringbuffer/src/main.c

```diff
@@ -39,20 +39,54 @@ RING_BUF_ITEM_DECLARE_POW2(ring_buf1, 8);
 #define DATA_MAX_SIZE 3
 #define POW 2
 
-
 /**
- * @brief test APIs of ring buffer
+ * @brief Test APIs of ring buffer
  *
- * @details define and initialize a ring buffer
+ * @details
+ * Test Objective:
+ * - define and initialize a ring buffer and
  * the ring buffer copy data out of the array by
  * ring_buf_item_put(), and then ring buffer data
  * is copied into the array by ring_buf_item_get()
  * return error when full/empty.
  *
- * @ingroup lib_ringbuffer_tests
+ * Testing techniques:
+ * - Interface testing
+ * - Dynamic analysis and testing
+ * - Equivalence classes and input partition testing
+ * - Structural test coverage(entry points,statements,branches)
  *
- * @see ring_buf_item_put(), ring_buf_item_get()
+ * Prerequisite Conditions:
+ * - Define and initialize a ringbuffer by using macro
+ * RING_BUF_ITEM_DECLARE_POW2
+ *
+ * Input Specifications:
+ * - N/A
+ *
+ * Test Procedure:
+ * -# defined an array with some data items that ready for being
+ * put.
+ * -# Do a while loop to put data items.
+ * -# Check if an error will be met when the ringbuffer is full.
+ * -# Get data items from the ringbuffer.
+ * -# Check if the data put are equal to the data got.
+ * -# Going on getting data from the ringbuffer.
+ * -# Check if an error will be met when the ringbuffer is empty.
  *
+ * Expected Test Result:
+ * - Data items that pushed will be equal to data items that got and
```

Comment:
It better to chnage it like this

Data items pushed shall be equal to what are gotten.